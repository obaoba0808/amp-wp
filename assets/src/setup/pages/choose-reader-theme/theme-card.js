
/**
 * WordPress dependencies
 */
import { useContext } from '@wordpress/element';
import { decodeEntities } from '@wordpress/html-entities';

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import './style.css';
import { __ } from '@wordpress/i18n';
import { Options } from '../../components/options-context-provider';

/**
 * A selectable card showing a theme in a list of themes.
 *
 * @param {Object} props Component props.
 * @param {string} props.description Theme description.
 * @param {string} props.homepage Link to view more information about the theme.
 * @param {string} props.name Theme name.
 * @param {string} props.slug Theme slug.
 */
export function ThemeCard( { description, homepage, screenshotUrl, slug, name } ) {
	const { options, updateOptions } = useContext( Options );
	const { reader_theme: readerTheme } = options || {};

	const id = `theme-card__${ slug }`;

	return (
		<li className={ `amp-wp-theme-card ${ readerTheme === slug ? 'amp-wp-theme-card--selected' : '' }` }>
			<label htmlFor={ id } className="amp-wp-theme-card__label">
				<img
					src={ screenshotUrl }
					alt={ name }
				/>
				<div className="amp-wp-theme-card__label-header">
					<input
						type="radio"
						id={ id }
						checked={ readerTheme === slug }
						onChange={ ( ) => {
							updateOptions( { reader_theme: slug } );
						} }
					/>
					<h2>
						{ decodeEntities( name ) }
					</h2>
				</div>

				<p>
					{ decodeEntities( description ) }
				</p>
			</label>
			<p className="amp-wp-theme-card__theme-link">
				<a href={ homepage } target="_blank" rel="noreferrer noopener">
					{ __( 'Learn more', 'amp' ) }
				</a>
			</p>
		</li>
	);
}

ThemeCard.propTypes = {
	description: PropTypes.string.isRequired,
	homepage: PropTypes.string.isRequired,
	screenshotUrl: PropTypes.string.isRequired,
	slug: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
};
