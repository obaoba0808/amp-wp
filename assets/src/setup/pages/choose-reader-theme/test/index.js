
/**
 * External dependencies
 */
import { create } from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { ThemeCard } from '../theme-card';
import { Providers } from '../../..';

describe( 'ThemeCard', () => {
	it( 'matches snapshot', () => {
		const wrapper = create(
			<Providers>
				<ThemeCard homepage="http://mysite.com" description="Theme card" screenshotUrl="http://screenshot.jpeg" slug="theme-slug" name="Theme name" />
			</Providers>,
		);
		expect( wrapper.toJSON() ).toMatchSnapshot();
	} );
} );
