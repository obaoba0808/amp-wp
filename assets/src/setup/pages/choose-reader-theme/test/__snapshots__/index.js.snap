// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeCard matches snapshot 1`] = `
<li
  className="amp-wp-theme-card "
>
  <label
    className="amp-wp-theme-card__label"
    htmlFor="theme-card__theme-slug"
  >
    <img
      alt="Theme name"
      src="http://screenshot.jpeg"
    />
    <div
      className="amp-wp-theme-card__label-header"
    >
      <input
        checked={false}
        id="theme-card__theme-slug"
        onChange={[Function]}
        type="radio"
      />
      <h2>
        Theme name
      </h2>
    </div>
    <p>
      Theme card
    </p>
  </label>
  <p
    className="amp-wp-theme-card__theme-link"
  >
    <a
      href="http://mysite.com"
      rel="noreferrer noopener"
      target="_blank"
    >
      Learn more
    </a>
  </p>
</li>
`;
